### PR TITLE
feat(hogql): Support `quantile` and parametric functions generally

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

[Parametric aggregated function](https://clickhouse.com/docs/en/sql-reference/aggregate-functions/parametric-functions) like `quantile` weren't supported in HogQL.

## Changes

Now they are. Only the basic `quantile` for now, and not e.g. `quantileExact`, to keep the surface area of HogQL small for now (and the `Exact` versions are more expensive to calculate).

## How did you test this code?

Bunch of tests.

### CODE REVIEW

File: posthog/hogql/printer.py [hunk 1]
```diff
@@ -439,30 +441,22 @@ def visit_call(self, node: ast.Call):
...
-                    f"Function '{node.name}' expects at most least {max_args} arguments. Passed {len(node.args)}."
+            validate_function_args(node.args, func_meta.min_args, func_meta.max_args, node.name)
+            if func_meta.min_params:
+                if node.params is None:
+                    raise HogQLException(f"Function '{node.name}' requires parameters in addition to arguments")
+                validate_function_args(
+                    node.params, func_meta.min_params, func_meta.max_params, node.name, argument_term="parameter"
                 )
 
             if self.dialect == "clickhouse":
```
Comment (Twixes) [inline posthog/hogql/printer.py:462]: Why are aggregate functions in a separate branch from the scalar ones?

**Comment by Twixes** [comment]
Bump @mariusandra

File: posthog/hogql/printer.py (hunk 1 above)
Comment (mariusandra) [reply posthog/hogql/printer.py:495]: I don't have a better answer than "legacy hack from the python AST days"

**Review by mariusandra — APPROVED** [review]
Looks good to me! 👍

File: posthog/hogql/printer.py (hunk 1 above)
Comment (Twixes) [reply posthog/hogql/printer.py:495]: Would be nice to consolidate… in another PR

**Pull request merged**